src/gdrdrv/gdrdrv.c

```diff
@@ -280,6 +281,20 @@ struct gdr_info {
 };
 typedef struct gdr_info gdr_info_t;
 
+static int gdrdrv_check_same_process(gdr_info_t *info, struct task_struct *tsk)
+{
+    int same_proc;
+    BUG_ON(0 == info);
+    BUG_ON(0 == tsk);
+    same_proc = (info->pid == task_pid(tsk))        // either exactly the same task
```

Comment:
Do you still need to check task_pid? Doesn't the task_tgid checking alone already cover the task_pid checking?

Reply:
I had the same idea, but somehow wanted to be on the safe side. I'll get rid of the redundant tracking and check.

Reply:
done